cmd/container-use/stdio_test.go

```diff
@@ -0,0 +1,359 @@
+	request.Params.Name = "environment_create"
+	request.Params.Arguments = map[string]any{
+		"environment_source": s.repoDir,
+		"title":              title,
+		"explanation":        explanation,
+	}
+
+	result, err := s.client.CallTool(ctx, request)
+	if err != nil {
+		return "", err
+	}
+
+	// Parse the environment ID from the result
+	if len(result.Content) > 0 {
+		if textContent, ok := result.Content[0].(mcp.TextContent); ok {
+			var envResponse struct {
+				ID string `json:"id"`
+			}
+			if err := json.Unmarshal([]byte(textContent.Text), &envResponse); err != nil {
+				return "", fmt.Errorf("failed to parse environment response (content: %q): %w", textContent.Text, err)
+			}
+			return envResponse.ID, nil
+		}
+	}
+
+	return "", fmt.Errorf("no valid response content found")
+}
+
+// FileRead reads a file from an environment via MCP
+func (s *MCPServerProcess) FileRead(envID, targetFile string) (string, error) {
+	ctx := context.Background()
+
+	request := mcp.CallToolRequest{}
+	request.Params.Name = "environment_file_read"
+	request.Params.Arguments = map[string]any{
+		"environment_source":      s.repoDir,
+		"environment_id":          envID,
+		"target_file":             targetFile,
+		"should_read_entire_file": true,
+		"explanation":             "Reading file for verification",
+	}
+
+	result, err := s.client.CallTool(ctx, request)
+	if err != nil {
+		return "", err
+	}
+
+	// Extract file content from result
+	if len(result.Content) > 0 {
+		if textContent, ok := result.Content[0].(mcp.TextContent); ok {
+			return textContent.Text, nil
+		}
+	}
+
+	return "", nil
+}
+
+// FileWrite writes a file to an environment via MCP
+func (s *MCPServerProcess) FileWrite(envID, targetFile, contents, explanation string) error {
+	ctx := context.Background()
+
+	request := mcp.CallToolRequest{}
+	request.Params.Name = "environment_file_write"
+	request.Params.Arguments = map[string]any{
+		"environment_source": s.repoDir,
+		"environment_id":     envID,
+		"target_file":        targetFile,
+		"contents":           contents,
+		"explanation":        explanation,
+	}
+
+	_, err := s.client.CallTool(ctx, request)
+	return err
+}
+
+// RunCommand executes a command in an environment via MCP
+func (s *MCPServerProcess) RunCommand(envID, command, explanation string) (string, error) {
+	ctx := context.Background()
+
+	request := mcp.CallToolRequest{}
+	request.Params.Name = "environment_run_cmd"
+	request.Params.Arguments = map[string]any{
+		"environment_source": s.repoDir,
+		"environment_id":     envID,
+		"command":            command,
+		"explanation":        explanation,
+	}
+
+	result, err := s.client.CallTool(ctx, request)
+	if err != nil {
+		return "", err
+	}
+
+	// Extract command output from result
+	if len(result.Content) > 0 {
+		if textContent, ok := result.Content[0].(mcp.TextContent); ok {
+			return textContent.Text, nil
+		}
+	}
+
+	return "", nil
+}
+
+// Test Cases
+
+func TestSharedRepositoryContention(t *testing.T) {
```

Comment:
🔧 **Maintainability (Low):** More descriptive skip message would help developers understand why the test is being skipped

```suggestion
if testing.Short() {
		t.Skip("Skipping slow E2E concurrency test")
	}
```